Tests/AlicerceTests/Analytics/AnalyticsTestCase.swift

```diff
@@ -0,0 +1,154 @@
+        let analyticsConfiguration = Analytics.Configuration(extraParameters: [
+            "1" : "1"
+        ])
+        
+        let analytics = Analytics(configuration: analyticsConfiguration)
+        
+        XCTAssertNotNil(analytics.parameters)
+        XCTAssertEqual(analytics.parameters?.count, 1)
+        
+        analytics.add(parameters: ["2" : "2"])
+        
+        XCTAssertEqual(analytics.parameters?.count, 2)
+        XCTAssertEqual(analytics.parameters!["1"] as! String, "1")
+        XCTAssertEqual(analytics.parameters!["2"] as! String, "2")
+    }
+    
+    func testAddTracker_WhenATrackerIsAdded_ItShouldSendEventsToThatTracker() {
+        let expectation = self.expectation(description: "Track appended tracker")
+
+        let page = Analytics.Page(name: "trackPage", parameters: nil)
+        let testableTracker = TestableTracker(expectation: expectation, page: page)
+        
+        let analytics = Analytics()
+        analytics.add(tracker: testableTracker)
+        
+        analytics.track(page: page)
+        
+        waitForExpectations(timeout: 30)
+    }
+    
+    func testTrackPage_WithoutGlobalParameters_ItShouldMatchTheOriginalPage() {
+        let expectation = self.expectation(description: "Track page to the testable tracker")
+        
+        let page = Analytics.Page(name: "trackablePage", parameters: ["1" : "1"])
+        let tracker = TestableTracker(expectation: expectation, page: page)
+        
+        let analytics = Analytics()
+        analytics.add(tracker: tracker)
+        
+        analytics.track(page: page)
+        
+        waitForExpectations(timeout: 30)
+    }
+    
+    func testTrackPage_WithGlobalParameters_ItShouldMatchAPageWithMoreParameters() {
+        let expectation = self.expectation(description: "Track page to the testable tracker")
+        
+        let trackerPage = Analytics.Page(name: "trackablePage", parameters: ["1" : "1", "2" : "2"])
+        let tracker = TestableTracker(expectation: expectation, page: trackerPage)
+        
+        let analyticsConfiguration = Analytics.Configuration(extraParameters: [
+            "1" : "1"
+            ])
+        
+        let analytics = Analytics(configuration: analyticsConfiguration)
+        analytics.add(tracker: tracker)
+        
+        let page = Analytics.Page(name: "trackablePage", parameters: ["2" : "2"])
+        analytics.track(page: page)
+        
+        waitForExpectations(timeout: 30)
+    }
+    
+    func testTrackEvent_WithoutGlobalParameters_ItShouldMatchOriginalEvent() {
+        let expectation = self.expectation(description: "Track page to the testable tracker")
+        
+        let event = Analytics.Event(name: "trackableEvent", parameters: ["1" : "1"])
+        let tracker = TestableTracker(expectation: expectation, event: event)
+        
+        let analytics = Analytics()
+        analytics.add(tracker: tracker)
+        
+        analytics.track(event: event)
+        
```

Comment:
You know in this case that the count should increase. if you use a different merge strategy, it would stay the same, or anything else. We should test behaviour of this API, not other intern logic.
I should not test internal logic, or neither know internally how this works.

If i send a dictionary with two values and then i send another with 3, they should be 5 if there is no repeated values. if there are repeated values, it can be 3 or 4, we know how many they will be, but depending on the merge strategy it can pick the value from the left dictionary or from the right, we don't know.